.github/workflows/ur-precommit.yml

```diff
@@ -0,0 +1,193 @@
+name: Unified Runtime Pre Commit
+# Note: this is the very first version of UR workflow.
+# It was pretty much copy-pasted from UR repository.
+# Over time it will be most likely integrated more into existing workflows.
+
+# Note: the trigger is copy-pasted from sycl-linux-precommit.yml - probably to be fine-tuned.
+on:
+  # We rely on "Fork pull request workflows from outside collaborators" -
+  # "Require approval for all outside collaborators" at
+  # https://github.com/intel/llvm/settings/actions for security.
+  pull_request:
+    branches:
+    - sycl
+    - sycl-rel-**
+    # Do not run builds if changes are only in the following locations
+    paths-ignore:
+    - '.github/ISSUE_TEMPLATE/**'
+    - '.github/CODEOWNERS'
+    - 'sycl/doc/**'
+    - 'sycl/gdb/**'
+    - 'clang/docs/**'
+    - '**.md'
+    - '**.rst'
+    - '.github/workflows/sycl-windows-*.yml'
+    - '.github/workflows/sycl-macos-*.yml'
+    - '.github/workflows/sycl-nightly.yml'
+    - '.github/workflows/sycl-rel-nightly.yml'
+    - 'devops/containers/**'
+    - 'devops/actions/build_container/**'
+
+concurrency:
+  #  Cancel a currently running workflow from the same PR, branch or tag.
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+permissions: read-all
+
+jobs:
+  detect_changes:
+    uses: ./.github/workflows/sycl-detect-changes.yml
+
+  # TODO: If UR is merged into llvm it will require changes:
+  # - 'detect_changes' should be required for all UR jobs
+  # - 'if' condition should be used, for all UR jobs, to check if UR is affected
+  #     (see example test_job's if)
+  # - test_job should be removed
+  #
+  test_job:
+    name: UR test job
+    needs: [detect_changes]
+    if: ${{ always() && !cancelled() && contains(needs.detect_changes.outputs.filters, 'ur') }}
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Check if UR is affected
+        run: |
+          echo "UR affected"
+          echo 'Filters set: ${{needs.detect_changes.outputs.filters}}'
+
+  source_checks:
+    name: Source Checks
+    needs: [detect_changes]
+    uses: ./.github/workflows/ur-source-checks.yml
+
+  level_zero:
+    name: Level Zero
+    needs: [source_checks]
+    uses: ./.github/workflows/ur-build-hw.yml
+    with:
+      adapter_name: L0
+      runner_name: UR_L0
+
+  level_zero_v2:
+    name: Level Zero V2
+    needs: [source_checks]
+    uses: ./.github/workflows/ur-build-hw.yml
+    with:
+      adapter_name: L0_V2
+      runner_name: UR_L0
+
+  level_zero_static:
```

Comment:
we could use a matrix for all of the adapters

Reply:
```
    strategy:
      fail-fast: false
      matrix:
        include:
```
? That's what we use in E2E tasks.

Reply:
right, we do the same thing with a matrix